engines/dfc_provider/spec/services/catalog_item_builder_spec.rb

```diff
@@ -28,4 +28,23 @@
       )
     end
   end
+
+  describe ".apply_stock" do
+    let(:item) { CatalogItemBuilder.catalog_item(variant) }
+
+    it "updates from on-demand to out-of-stock" do
+      variant.save!
+      variant.on_demand = true
+      variant.on_hand = -3
+
+      item.stockLimitation = 0
+
+      expect {
+        CatalogItemBuilder.apply_stock(item, variant)
+        variant.save!
+      }
+        .to change { variant.on_demand }.to(false)
+        .and change { variant.on_hand }.to(0)
+    end
+  end
```

Comment:
Might be worth testing the inverse also?

Reply:
Maybe. But maybe not needed?

Reply:
😁 🤷